### ISSUE TITLE
Cross-origin issue while accessing to iframe content

### ISSUE THREAD

**Issue by ninja-pinky** [Typeform/embed#441]
I want to access to contentWindow of the iframe but I can't access to the iframe element because cross-origin restrictions.

```
// run after loading new Question
function changed() {
        console.log('New Question')
        var iframe = document.querySelector(".tf-v1-widget");
        console.log(iframe.getElementsByTagName("iframe")[0].contentWindow.document)
}
```
![image](https://user-images.githubusercontent.com/73223166/148749973-8ef703fe-494d-469f-b156-4619dced4930.png)

Is there any way via Typeform Dashboard to disable CORS-blocking restrictions for specific website or any other solution to solove this problem?

**Comment by mathio** [comment]
Hello @ninja-pinky for security reasons this is not possible. 

We use [postMessage](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage) for communication between the form itself and embed library for functionality we intentionally want to expose (eg. [callbacks](https://github.com/Typeform/embed/tree/main/packages/embed#callbacks)). 

If you want to modify your typeform via DOM manipulation you can do so by injecting scripts via [Google Tag Manager integration](https://www.typeform.com/help/a/add-google-tag-manager-to-a-typeform-360029583611/) however I would strongly advise not to, since your modifications might break the form or stop working at any time because we do not support such customizations.

What are you trying to achieve? Maybe we or [our community](https://community.typeform.com/) will be able to help find a long-term solution?

**Comment by ninja-pinky** [comment]
Thanks for your reply.

I have a form with multiple steps but the steps have different heights so I need a solution to avoid the scrollbar while embedding the form. 
The idea is to get the height of every step(by accessing to document in iframe with iframe.contentWindow.document) while question changed (OnQuestionChanged event) and change the height of iframe with the height of the current step.

```
<div 
   data-tf-widget="<form-id>"
   data-tf-on-question-changed="changed"
   style="width:100%;height:[will be changed every time after question changed];">
</div>
<script src="//embed.typeform.com/next/embed.js"></script>
```

is there any other solution to solve this problem?

**Comment by mathio** [comment]
@ninja-pinky I am afraid that this is not something we support right now. Maybe we could send the form height as payload to the OnQuestionChanged event? 🤔

**Comment by abdes-zakari** [comment]
Hi @mathio thank you for your suggestion.

Yeah I think that will be a good idea if we have the form height as payload in the OnQuestionChanged event.

I'm waiting for this.

**Comment by ninja-pinky** [comment]
Hi @mathio  thank you for your suggestion.

Yeah I think that will be a good idea if we have the form height as payload in the OnQuestionChanged event.

**Comment by mathio** [comment]
Now you can use `onHeightChanged` callback. If you want to resize widget embed you can use `autoResize` option.

This issue was resolved in #463
